Formula/onlykey-agent.rb

```diff
@@ -0,0 +1,213 @@
+  depends_on "openssl@1.1"
+
+  depends_on "python@3.10"
+
+  # six come from brew create --python adding it as a resource
+  depends_on "six"
+
+  resource "aenum" do
+    url "https://files.pythonhosted.org/packages/63/6c/a71e18de7c651f384b328be6bccadbbd472aca62f547c1a307b9388d03ca/aenum-3.1.11.tar.gz"
+    sha256 "aed2c273547ae72a0d5ee869719c02a643da16bf507c80958faadc7e038e3f73"
+  end
+
+  resource "hidapi" do
+    url "https://files.pythonhosted.org/packages/ef/72/54273f701c737ae5f42d9c0adf641912d20eb955c75433f1093fa509bcc7/hidapi-0.12.0.post2.tar.gz"
+    sha256 "8ebb2117be8b27af5c780936030148e1971b6b7fda06e0581ff0bfb15e94ed76"
+  end
+
+  resource "backports.shutil_which" do
+    url "https://files.pythonhosted.org/packages/a0/22/51b896a4539f1bff6a7ab8514eb031b9f43f12bff23f75a4c3f4e9a666e5/backports.shutil_which-3.5.2.tar.gz"
+    sha256 "fe39f567cbe4fad89e8ac4dbeb23f87ef80f7fe8e829669d0221ecdb0437c133"
+  end
+
+  resource "bech32" do
+    url "https://files.pythonhosted.org/packages/ab/fe/b67ac9b123e25a3c1b8fc3f3c92648804516ab44215adb165284e024c43f/bech32-1.2.0.tar.gz"
+    sha256 "7d6db8214603bd7871fcfa6c0826ef68b85b0abd90fa21c285a9c5e21d2bd899"
+  end
+
+  resource "certifi" do
+    url "https://files.pythonhosted.org/packages/cc/85/319a8a684e8ac6d87a1193090e06b6bbb302717496380e225ee10487c888/certifi-2022.6.15.tar.gz"
+    sha256 "84c85a9078b11105f04f3036a9482ae10e4621616db313fe045dd24743a0820d"
+  end
+
+  resource "cffi" do
+    url "https://files.pythonhosted.org/packages/00/9e/92de7e1217ccc3d5f352ba21e52398372525765b2e0c4530e6eb2ba9282a/cffi-1.15.0.tar.gz"
+    sha256 "920f0d66a896c2d99f0adbb391f990a84091179542c205fa53ce5787aff87954"
+  end
+
+  resource "charset-normalizer" do
+    url "https://files.pythonhosted.org/packages/56/31/7bcaf657fafb3c6db8c787a865434290b726653c912085fbd371e9b92e1c/charset-normalizer-2.0.12.tar.gz"
+    sha256 "2857e29ff0d34db842cd7ca3230549d1a697f96ee6d3fb071cfa6c7393832597"
+  end
+
+  resource "click" do
+    url "https://files.pythonhosted.org/packages/59/87/84326af34517fca8c58418d148f2403df25303e02736832403587318e9e8/click-8.1.3.tar.gz"
+    sha256 "7682dc8afb30297001674575ea00d1814d808d6a36af415a82bd481d37ba7b8e"
+  end
+
+  resource "ConfigArgParse" do
+    url "https://files.pythonhosted.org/packages/16/05/385451bc8d20a3aa1d8934b32bd65847c100849ebba397dbf6c74566b237/ConfigArgParse-1.5.3.tar.gz"
+    sha256 "1b0b3cbf664ab59dada57123c81eff3d9737e0d11d8cf79e3d6eb10823f1739f"
+  end
+
+  resource "cryptography" do
+    url "https://files.pythonhosted.org/packages/51/05/bb2b681f6a77276fc423d04187c39dafdb65b799c8d87b62ca82659f9ead/cryptography-37.0.2.tar.gz"
+    sha256 "f224ad253cc9cea7568f49077007d2263efa57396a2f2f78114066fd54b5c68e"
+  end
+
+  resource "Cython" do
+    url "https://files.pythonhosted.org/packages/d4/ad/7ce0cccd68824ac9623daf4e973c587aa7e2d23418cd028f8860c80651f5/Cython-0.29.30.tar.gz"
+    sha256 "2235b62da8fe6fa8b99422c8e583f2fb95e143867d337b5c75e4b9a1a865f9e3"
+  end
```

Comment:
Or even `depends_on "libcython"`

Reply:
I tried replacing it with `depends_on "libcython"` and brew test failed with the following on the ubuntu after a full build with the following:

```
#13 4.064   Minitest::Assertion: Expected /try unplugging and replugging your device/ to match "Traceback (most recent call last):
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/bin/onlykey-agent\", line 33, in <module>
    sys.exit(load_entry_point('onlykey-agent==1.1.14', 'console_scripts', 'onlykey-agent')())
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/bin/onlykey-agent\", line 25, in importlib_load_entry_point
    return next(matches).load()
  File \"/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/importlib/metadata/__init__.py\", line 171, in load
    module = import_module(match.group('module'))
  File \"/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/importlib/__init__.py\", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File \"<frozen importlib._bootstrap>\", line 1050, in _gcd_import
  File \"<frozen importlib._bootstrap>\", line 1027, in _find_and_load
  File \"<frozen importlib._bootstrap>\", line 1006, in _find_and_load_unlocked
  File \"<frozen importlib._bootstrap>\", line 688, in _load_unlocked
  File \"<frozen importlib._bootstrap_external>\", line 883, in exec_module
  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/bin/onlykey_agent.py\", line 1, in <module>
    import libagent.gpg
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/libagent/gpg/__init__.py\", line 22, in <module>
    import pkg_resources
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 3260, in <module>
    def _initialize_master_working_set():
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 3234, in _call_aside
    f(*args, **kwargs)
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 3272, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 581, in _build_master
    ws.require(__requires__)
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 909, in require
    needed = self.resolve(parse_requirements(requirements))
  File \"/home/linuxbrew/.linuxbrew/Cellar/onlykey-agent/1.1.14/libexec/lib/python3.10/site-packages/pkg_resources/__init__.py\", line 795, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'Cython>=0.23.4' distribution was not found and is required by onlykey
".
```

I'll try replacing it with `depends_on "cython"` instead...

Reply:
Looks like `depends_on "cython"` fails since that formula isn't updated with python 3.10 which I was [advised to update this package to in a prior comment on this PR](https://github.com/Homebrew/homebrew-core/pull/104096#discussion_r902676853). So some options to move forward:

1. Let `resource "Cython"` that `brew create` added stand 
2. Revert this package's dependency back to python@3.9 that `brew create` used
3. Attempt to update the cython formula to depend on pthon@3.10 and merge that before merging this one. I found the 3.9 bump at f23cda5147ab938d7977e34884d9535845b5353e and can make a similar change for 3.10 but and submit that - but admittedly I am not clear on the consequences of doing that.

Advice?

Reply:
`cython` formula shouldn't be used unless it needs the executables (which is rare).

`libcython` formula (and probably `cython`) will probably fail due to them being keg-only, so the libraries cannot be found without some extra work. One option is to add a `.pth` file to virtualenv site-packages to pick up keg-only libraries.

Otherwise, we would need to use resource.

Reply:
Thank you @cho-m 🙇‍♂️ I'll move forward with resources for now. If maintainers have a concern please reply and I'd appreciate a bit more direction on the .pth/virtualenv site-packages suggestion if that is necessary as I'm not familiar with that.

Reply:
> One option is to add a `.pth` file to virtualenv site-packages to pick up keg-only libraries.

I think `brew` [handles something like this](https://github.com/Homebrew/brew/blob/976681134384abf0466bbf3ce4acdb880a262d84/Library/Homebrew/language/python.rb#L136-L151).

```ruby
        # Find any Python bindings provided by recursive dependencies
        formula_deps = formula.recursive_dependencies
        pth_contents = formula_deps.map do |d|
          next if d.build? || d.test?
          # Do not add the main site-package provided by the brewed
          # Python formula, to keep the virtual-env's site-package pristine
          next if python_names.include? d.name

          dep_site_packages = Formula[d.name].opt_prefix/Language::Python.site_packages(python)
          next unless dep_site_packages.exist?

          "import site; site.addsitedir('#{dep_site_packages}')\n"
        end.compact
        unless pth_contents.empty?
          (venv_root/Language::Python.site_packages(python)/"homebrew_deps.pth").write pth_contents.join
        end
```

It just needs a little adjustment.

Reply:
@cho-m or @carlocab I'm spinning a bit on this and have some questions:

1. I should run this code from the formula's `def install` block before `virtualenv_install_with_resources`, right?
2. How do I get the path to _virtualenv site-packages_ (mentioned above by @cho-m) where I need write the .pth?
3. What should the .pth file's name be exactly? `homebrew-onlykey-agent-dependencies.pth` )? I am basing that on https://docs.brew.sh/Python-for-Formula-Authors#dependencies-1.
4. How exactly should I get the path to libcython (to use with `site.addsitedir`) to write into the .pth file?

Reply:
If you want an example, I did do this for `alot` to use `notmuch` Python bindings
https://github.com/Homebrew/homebrew-core/blob/bc8a1f204ae253d407782691d27707eb8a1bc306/Formula/alot.rb#L101-L104

Before or after `virtualenv_install_with_resources` depends on if it is needed while building other dependencies or not. In `alot`, it is only a runtime dependency so could be created afterward.

Reply:
I have resolved this in 56e444cb9c64135197dead6c696e9bd8cb37cadc which will replace `resource "Cython"` with `depends_on "libcython"`. Thank you for your help @cho-m !